File: source/clibCore/Controllers/errorController.vb

Option Explicit On
Option Strict On
'
Imports System.Text.RegularExpressions
'
Namespace Contensive.Core.Controllers
    '
    '====================================================================================================
    ''' <summary>
    ''' static class controller
    ''' </summary>
    Public Class errorController
        Implements IDisposable
        '
        ' ----- constants
        '
        'Private Const invalidationDaysDefault As Double = 365
        '
        '==========================================================================
        '   Add on to the common error message
        '==========================================================================
        '
        Public Shared Sub error_AddUserError(cpCore As coreClass, ByVal Message As String)
            If (InStr(1, cpCore.debug_iUserError, Message, vbTextCompare) = 0) Then
                cpCore.debug_iUserError = cpCore.debug_iUserError & cr & "<li class=""ccError"">" & genericController.encodeText(Message) & "</LI>"
            End If
        End Sub
        '
        '==========================================================================
        '   main_Get The user error messages
        '       If there are none, return ""
        '==========================================================================
        '
        Public Shared Function error_GetUserError(cpcore As coreClass) As String
            error_GetUserError = genericController.encodeText(cpcore.debug_iUserError)
            If error_GetUserError <> "" Then
                error_GetUserError = "<ul class=""ccError"">" & genericController.htmlIndent(error_GetUserError) & cr & "</ul>"
                error_GetUserError = UserErrorHeadline & "" & error_GetUserError
                cpcore.debug_iUserError = ""
            End If
        End Function

        '
        '==========================================================================================
        ''' <summary>
        ''' return an html ul list of each eception produced during this document.
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Shared Function getDocExceptionHtmlList(cpcore As coreClass) As String
            Dim returnHtmlList As String = ""
            Try
                If Not cpcore.errList Is Nothing Then
                    If cpcore.errList.Count > 0 Then
                        For Each exMsg As String In cpcore.errList
                            returnHtmlList &= cr2 & "<li class=""ccExceptionListRow"">" & cr3 & cpcore.htmlDoc.html_convertText2HTML(exMsg) & cr2 & "</li>"
                        Next
                        returnHtmlList = cr & "<ul class=""ccExceptionList"">" & returnHtmlList & cr & "</ul>"
                    End If
                End If
            Catch ex As Exception
                Throw (ex)
            End Try
            Return returnHtmlList
        End Function

        '
        '====================================================================================================
#Region " IDisposable Support "
        '
        ' this class must implement System.IDisposable
        ' never throw an exception in dispose
        ' Do not change or add Overridable to these methods.
        ' Put cleanup code in Dispose(ByVal disposing As Boolean).
        '====================================================================================================
        '
        Protected disposed As Boolean = False
        '
        Public Overloads Sub Dispose() Implements IDisposable.Dispose
            ' do not add code here. Use the Dispose(disposing) overload
            Dispose(True)
            GC.SuppressFinalize(Me)
        End Sub
        '
        Protected Overrides Sub Finalize()
            ' do not add code here. Use the Dispose(disposing) overload
            Dispose(False)
            MyBase.Finalize()
        End Sub
        '
        '====================================================================================================
        ''' <summary>
        ''' dispose.
        ''' </summary>
        ''' <param name="disposing"></param>
        Protected Overridable Overloads Sub Dispose(ByVal disposing As Boolean)
            If Not Me.disposed Then
                Me.disposed = True
                If disposing Then
                    'If (cacheClient IsNot Nothing) Then
                    '    cacheClient.Dispose()
                    'End If
                End If
                '
                ' cleanup non-managed objects
                '
            End If
        End Sub
#End Region
    End Class
    '
End Namespace

File: source/Processor/Controllers/bak/errorController.cs


using System.Text.RegularExpressions;
// 

namespace Controllers {
    
    // 
    // ====================================================================================================
    // '' <summary>
    // '' static class controller
    // '' </summary>
    public class errorController : IDisposable {
        
        // 
        //  ----- constants
        // 
        // Private Const invalidationDaysDefault As Double = 365
        // 
        // ==========================================================================
        //    Add on to the common error message
        // ==========================================================================
        // 
        public static void error_AddUserError(coreClass cpCore, string Message) {
            if (((cpCore.doc.debug_iUserError.IndexOf(Message, 0, System.StringComparison.OrdinalIgnoreCase) + 1) 
                        == 0)) {
                cpCore.doc.debug_iUserError = (cpCore.doc.debug_iUserError 
                            + (cr + ("<li class=\"ccError\">" 
                            + (genericController.encodeText(Message) + "</LI>"))));
            }
            
        }
        
        // 
        // ==========================================================================
        //    main_Get The user error messages
        //        If there are none, return ""
        // ==========================================================================
        // 
        public static string error_GetUserError(coreClass cpcore) {
            error_GetUserError = genericController.encodeText(cpCore.doc.debug_iUserError);
            if ((error_GetUserError != "")) {
                error_GetUserError = ("<ul class=\"ccError\">" 
                            + (genericController.htmlIndent(error_GetUserError) 
                            + (cr + "</ul>")));
                error_GetUserError = (UserErrorHeadline + ("" + error_GetUserError));
                cpCore.doc.debug_iUserError = "";
            }
            
        }
        
        // 
        // ==========================================================================================
        // '' <summary>
        // '' return an html ul list of each eception produced during this document.
        // '' </summary>
        // '' <returns></returns>
        // '' <remarks></remarks>
        public static string getDocExceptionHtmlList(coreClass cpcore) {
            string returnHtmlList = "";
            try {
                if (!(cpcore.doc.errList == null)) {
                    if ((cpcore.doc.errList.Count > 0)) {
                        foreach (string exMsg in cpcore.doc.errList) {
                            (cr2 + ("<li class=\"ccExceptionListRow\">" 
                                        + (cr3 
                                        + (cpcore.html.convertTextToHTML(exMsg) 
                                        + (cr2 + "</li>")))));
                        }
                        
                        returnHtmlList = (cr + ("<ul class=\"ccExceptionList\">" 
                                    + (returnHtmlList 
                                    + (cr + "</ul>"))));
                    }
                    
                }
                
            }
            catch (Exception ex) {
                throw ex;
            }
            
            return returnHtmlList;
        }
        
        // 
        //  this class must implement System.IDisposable
        //  never throw an exception in dispose
        //  Do not change or add Overridable to these methods.
        //  Put cleanup code in Dispose(ByVal disposing As Boolean).
        // ====================================================================================================
        // 
        protected bool disposed = false;
        
        public void Dispose() {
            //  do not add code here. Use the Dispose(disposing) overload
            this.Dispose(true);
            GC.SuppressFinalize(this);
        }
        
        // 
        protected override void Finalize() {
            //  do not add code here. Use the Dispose(disposing) overload
            this.Dispose(false);
            base.Finalize();
        }
        
        // 
        // ====================================================================================================
        // '' <summary>
        // '' dispose.
        // '' </summary>
        // '' <param name="disposing"></param>
        protected virtual void Dispose(bool disposing) {
            if (!this.disposed) {
                this.disposed = true;
                if (disposing) {
                    // If (cacheClient IsNot Nothing) Then
                    //     cacheClient.Dispose()
                    // End If
                }
                
                // 
                //  cleanup non-managed objects
                // 
            }
            
        }
    }
}